bfd/elfxx-riscv.c

```diff
@@ -1159,8 +1159,17 @@ static bfd_boolean
 riscv_ext_dont_care_version (const char *subset)
 {
   if (strcmp (subset, "g") == 0
+      || strcmp (subset, "k") == 0
       || strcmp (subset, "zicsr") == 0
-      || strcmp (subset, "zifencei") == 0)
+      || strcmp (subset, "zifencei") == 0
+      || strcmp (subset, "zkg") == 0
+      || strcmp (subset, "zkb") == 0
+      || strcmp (subset, "zkr") == 0
+      || strcmp (subset, "zkne") == 0
+      || strcmp (subset, "zknd") == 0
+      || strcmp (subset, "zknh") == 0
+      || strcmp (subset, "zksed") == 0
+      || strcmp (subset, "zksh") == 0)
```

Comment:
Yeah k and it's sub-extensions should have a draft version.  We won't output the extensions in the output ISA string, if you list them in the riscv_ext_dont_care_version. The function used to deal with the special extensions like zicsr and zifencei. Since they are extracted from i in the ISA spec v2.1, we should add them as the implicit extensions when the version of i is less than v2.1, but in the newer spec they are explicit. For example, add zicsr as implicit means we allow to use the related csrr instructions when only enable i v2.0, but we won't output them to the output ISA string since they don't have a default versions. The g-ext is another special case that it's version doesn't mean anything.